src/services/impl/FileLogger.hh

```diff
@@ -37,9 +37,19 @@ public:
             return;
 
         // write a timestamp
-        auto tNow = ServiceLocator::Get<IClock>().Now();
-        auto tMilliseconds = static_cast<unsigned int>(std::chrono::time_point_cast<std::chrono::milliseconds>(tNow).time_since_epoch().count() % 1000);
-        auto tTime = std::chrono::system_clock::to_time_t(tNow);
+        time_t tTime;
+        unsigned int tMilliseconds;
+        if (ServiceLocator::Exists<IClock>())
+        {
+            auto tNow = ServiceLocator::Get<IClock>().Now();
+            tMilliseconds = static_cast<unsigned int>(std::chrono::time_point_cast<std::chrono::milliseconds>(tNow).time_since_epoch().count() % 1000);
+            tTime = std::chrono::system_clock::to_time_t(tNow);
+        }
+        else
+        {
+            tTime = time(nullptr);
+            tMilliseconds = 0;
+        }
```

Comment:
prevents exception using RA_LOG before the IClock is registered or after it's been destroyed. I fixed the issue that caused me to run into this, so it may not be necessary in the long-term.